### PR TITLE
Making build pass for embedded platforms

### LINKED ISSUE #11: Currently won't properly compile for "embedded"
If neither `_WIN32` nor `unix` are defined, the following compilation error occurs:

```
libcester/include/exotic/cester.h:3199:13: error: label 'end_sub_process' used but not defined
 3199 |             goto end_sub_process;
      |             ^~~~
```

The label `end_sub_process` is only defined for the `_WIN32` or `unix` case, but not in the 3rd case fallback.

I guess this block of code needs to be disabled in this situation.

### PR DESCRIPTION
**Pull request by nicolasnoble**
Ought to fix #11.

### CODE REVIEW

File: include/exotic/cester.h [hunk 1]
```diff
@@ -3181,6 +3181,7 @@ static __CESTER_INLINE__ void cester_run_test(TestInstance *test_instance, TestC
 #else
         #pragma message("Isolated tests not supported in this environment. The tests will be run on the main process")
         last_status = cester_run_test_no_isolation(test_instance, a_test_case, index);
+#define CASTER_NO_SUBPROCESS 1
```
Comment (Thecarisma) [inline include/exotic/cester.h:3184]: The macro should start with CESTER_, not CASTER_

File: include/exotic/cester.h [hunk 2]
```diff
@@ -3194,10 +3195,12 @@ static __CESTER_INLINE__ void cester_run_test(TestInstance *test_instance, TestC
         last_status = superTestInstance.current_execution_status;
 #ifdef __STDC_VERSION__
 #ifndef __cplusplus
+#ifndef CASTER_NO_SUBPROCESS
```
Comment (Thecarisma) [inline include/exotic/cester.h:3198]: The macro should start with CESTER_, not CASTER_

**Comment by Thecarisma** [comment]
Hello Nicolas, Thanks for submitting the PR I was able to recreate the error with the following snippet

```c
#include <exotic/cester.h>
#undef unix
#undef _WIN32

CESTER_TEST(quick_test, test_instance, 
    cester_assert_true(1);
)
```

I drop some comments on the file to fix some typos, after that the PR should be good to merge

**Comment by nicolasnoble** [comment]
Ooops, sorry about the typos :)

File: include/exotic/cester.h (hunk 2 above)
Comment (nicolasnoble) [reply include/exotic/cester.h]: Done.

File: include/exotic/cester.h (hunk 1 above)
Comment (nicolasnoble) [reply include/exotic/cester.h]: Done.

**Pull request merged**